### PR TITLE
Replace e2e-harness test with go test

### PR DESCRIPTION
**Pull request by sslavic**
Towards giantswarm/giantswarm#4606

### CODE REVIEW

**Review by kopiczko — APPROVED** [review]

**Comment by kopiczko** [comment]
Sorry I'm retarded - I confused PRs :D

**Comment by xh3b4sd** [comment]
Happened to me already as well. 😄

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,7 +26,13 @@ e2eTest: &e2eTest
...
         name: Run test
         command: |
-          E2E_KUBECONFIG=/e2e/kubeconfig ./e2e-harness test --test-dir=${TEST_DIR}
+          docker run --rm --network host \
+            -v $(pwd)/${TEST_DIR}:/e2e \
+            -v $(pwd):/go/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME} \
+            -e E2E_KUBECONFIG=/e2e/kubeconfig \
+            -e CIRCLE_SHA1=${CIRCLE_SHA1} \
+            -w /go/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME} \
+            quay.io/giantswarm/golang:1.12.6 go test -v -tags k8srequired ./${TEST_DIR}/
```
Comment (xh3b4sd) [inline .circleci/config.yml]: Who is actually going to update all the go versions at some point?

Comment (sslavic) [reply .circleci/config.yml]: Good point. Quick solution could be to use tag 1, so it gets auto updated with every new 1.x.y release, at least for the next build. Additionally we can put this job/steps into a reusable orb command (either in existing architect or some new orb). @xh3b4sd @kopiczko what do you think?

Comment (sslavic) [reply .circleci/config.yml]: Switched to golang:1 via https://github.com/giantswarm/kubernetes-kube-state-metrics/pull/52

Comment (MarcelMue) [reply .circleci/config.yml]: I think pawel had the right idea there:
This whole block could be part of an orb, so we have a single place for this go version?
I think that should be possible.

Comment (kopiczko) [reply .circleci/config.yml]: Yeah making it an orb is a way to go IMO. I want to do this as a part of releng SWH.

**Pull request merged**